lec15-git-projects.Rmd

```diff
@@ -0,0 +1,415 @@
+---
+title: "Reproducible science with Git and GitHub"
+author: Ahmed Hasan
+---
+
+## Lesson preamble
+
+> ### Lesson objectives:
+>
+> - Understand basic bash commands
+> - Practice using Git commands at the command line
+> - Learn how and when to use branches
+> - Learn how to effectively use Pull Requests
+> - Learn how to work with forks
+>
+> ### Lesson outline:
+>
+> - Intro to Git (50 min)
+>       - Bash Primer (15 min)
+>       - Getting started with Git (5 min)
+>       - Version control with Git (30 min)
+> - Collaborating with GitHub (50 min)
+>       - Creating a fork + GitHub setup (5 min)
+>       - Branches and pull requests (40 min)
+>       - Issues (5 min)
+>
+-----
+
+## Intro to Git
+
+Git is a version control system that tracks changes in files. Although
+it is primarily used for software development, Git can be used to track
+changes in any files, and is an indispensable tool for collaborative projects.
+
+Git is often used via [GitHub](http://www.github.com), a website that
+hosts Git repositories. GitHub therefore facilitates collaboration
+by allowing for code review and other useful features. In the spirit of 
+'working open' and ensuring scientific reproducibility, it has also become increasingly 
+common for scientists to upload scripts and related files to GitHub for others to use.
+
+In this lesson, we'll be covering some basic Git commands, before
+moving on to using GitHub for the upcoming group projects.
+The members of group will be given admin access to a pre-made repository. 
+
+### Command line Git
+
+#### Setup and Installation
+
+Git is primarily used through a _bash interpreter_, also known as a 'bash shell', 
```

Comment:
@joelostblom would like to point out that zsh is definitely not a bash interpreter.
Anyway, I think this sentence is backwards; git is primarily used on the command line, of which bash interpreter/shell is one implementation.